datafusion/core/tests/sql/decimal.rs

```diff
@@ -879,37 +296,3 @@ async fn decimal_null_array_scalar_comparison() -> Result<()> {
     assert_eq!(&DataType::Boolean, actual[0].column(0).data_type());
```

Comment:
`decimal_null_scalar_array_comparison` and `decimal_null_array_scalar_comparison` would also be good candidates to be moved to sqllogictest. It would properly check that the results are `NULL`. But there is currently no way to differentiate if its a decimal null or a boolean null.

Reply:
I wonder if you could use `arrow_typeof` to do the check

```sql
❯ select null <= a from (values (1.1::decimal)) as t(a);
+-----------+
| NULL <= a |
+-----------+
|           |
+-----------+
1 row in set. Query took 0.029 seconds.
❯ select arrow_typeof(null <= a) from (values (1.1::decimal)) as t(a);
+------------------------+
| arrowtypeof(NULL <= a) |
+------------------------+
| Boolean                |
+------------------------+
1 row in set. Query took 0.006 seconds.
```

Reply:
Thank you for the hint, `arrow_typeof` works quite well! https://github.com/apache/arrow-datafusion/pull/5086/commits/8295aa5f7a138d1d44d3c05c436b88ef0b82140c